logbook-servlet/src/main/java/org/zalando/logbook/servlet/LocalResponse.java

```diff
@@ -106,42 +114,54 @@ public PrintWriter getWriter() throws IOException {
         return body;
     }
 
-    private final class TeeServletOutputStream extends ServletOutputStream {
+    private static final class TeeServletOutputStream extends ServletOutputStream {
 
-        private final OutputStream original;
+        private final DataOutput output;
+        private final LazyStream original;
 
-        private TeeServletOutputStream() throws IOException {
-            this.original = LocalResponse.super.getOutputStream();
+        private TeeServletOutputStream(final DataOutput output,
+                final LazyStream original) {
```

Comment:
Then maybe rename it to 
```java
   @FunctionalInterface
    interface StreamSupplier {
        OutputStream get() throws IOException;
    }
```
since there is nothing lazy happening